lib/algolia/index.rb

```diff
@@ -409,7 +409,9 @@ def delete_by_query(query, params = nil)
       loop do
         res = search(query, params)
         break if res['hits'].empty?
-        res = delete_objects(res['hits'].map { |h| h['objectID'] })
+        ids = res['hits'].map { |h| h['objectID'] }
+        res = delete_objects(ids)
+        break if ids.size != 1000 # no need to wait_task if we had less than 1000 objects matching; we won't get more after anyway.
```

Comment:
I think the problem here is that if you don't wait, the next `search()` calls might return the same object ids, thus flooding the API with duplicates delete calls.